Automattic/woocommerce-services
# [WCS in WPCOM] Refactor the client so the React app is entirely REST-driven

Fixes partially #995

For this first round, I've decided to focus on the `Packaging Manager` and the `Payment Selector` boxes, since they are the first that are going to be incorporated to Calypso.

The objective is to have those "sub-applications" be as independent of their WP-Admin context as possible. They must be able to initialize themselves completely using just REST endpoints.

This PR is big, I recommend reviewing it by commit (they are independent and self-contained). If you don't understand the "why" of some of these decision, please ask and I'll clarify (and probably add comments).

Since this is a partial refactor, there are some things I couldn't fully remove, such as `save-form.js`.

To test:
* Go to the Packaging Manager.
* Notice that it doesn't load instantly anymore, now it has a spinner that last for a second or so.
* Go to the Shipping Labels settings page (the Payment selector thingy).
* Notice that it doesn't load instantly anymore, now it has a spinner that last for a second or so.
* Check that, apart from that, everything works as usual. That includes the other parts of WCS, such as buying a label, configuring USPS, self-help, etc.

@olaolusoga You may want to take a look at this.

This is the "Label Settings" when it's loading:
![progress-fetch](https://cloud.githubusercontent.com/assets/1715800/26077677/cda53ce2-39b4-11e7-958d-3ca7903dd602.png)

And this is if it couldn't load its settings (shouldn't happen almost ever):
![error-fetch](https://cloud.githubusercontent.com/assets/1715800/26077652/b3a49608-39b4-11e7-92c6-9bb282f7526f.png)

The "Packaging Manager" follows the same pattern, spinner & disabled button when it's loading, generic message & disabled button on error.
